### PR TITLE
Fix audit issue logging by default peer address

### PR DESCRIPTION
**Pull request by pando85**
By default log format include remote address that is taken from headers.
This is very easy to replace making log untrusted.

Changing default log format value `%a` to peer address we are getting
this trusted data always. Also, remote address option is maintianed and
relegated to `%{r}a` value.

Related  kanidm/kanidm#191.

### CODE REVIEW

**Comment by robjtede** [comment]
So if I have this right, you take issue with the fact that the current %a replacement uses the derived "Client IP" which could be taken from X-Forwarded-For headers which are obviously easy to replace if your proxy is misconfigured (or you don't have a proxy and the header is being sent manually).

There's a couple bits of terminology I want to nail down, document, and be strict to use in the right places.
- Client Address
- Peer Address
- Remote Address

Could you elaborate on the above so there is no confusion in reviewing? Might be worth adding doc comments to the ConnectionInfo fields to describe when and what they are populated with.

**Comment by pando85** [comment]
@robjtede thanks for your comment!
I tried to be more clear about IP meanings choosing Nginx style names.

File: src/middleware/logger.rs [hunk 1]
```diff
@@ -72,12 +72,21 @@ use crate::HttpResponse;
...
 ///
+/// `%{r}a`  Real IP remote address **\***
+///
 /// `%{FOO}i`  request.headers['FOO']
 ///
 /// `%{FOO}o`  response.headers['FOO']
 ///
 /// `%{FOO}e`  os.environ['FOO']
 ///
+/// ### **\* security warning**
```
Comment (JohnTitor) [inline src/middleware/logger.rs]:
```suggestion
/// # Security
```
We usually use this header.

**Review by JohnTitor — COMMENTED** [review]
I think this is a reasonable change, could you also update the changelog?

**Comment by pando85** [comment]
@JohnTitor both done.

Must I rebase all changes in one?

**Comment by pando85** [comment]
Any update?

**Review by JohnTitor — APPROVED** [review]
I missed this, sorry. Thanks!

**Pull request merged**